15/umbraco-cms/fundamentals/setup/server-setup/running-umbraco-in-docker.md

```diff
@@ -0,0 +1,100 @@
+# Running Umbraco in Docker
+
+Exactly how you choose to compose your Dockerfile will depend on your project specific needs. This section is not intended as a comprehensive guide, rather as an overview of topics to be aware of when hosting in Docker.
+
+## What is Docker
+
+Docker is a platform for developing, shipping, and running applications in containers. Multiple services exist for hosting these containers.  For more information, [refer to the official Docker documentation](https://docs.docker.com/)
+
+## The Docker file system
+
+By default, files created inside a container are written to an ephemeral, writable container layer. 
+This means that the files don't persist when the container is removed, and it's challenging to get files out of the container. Additionally, this writable layer is not suitable for performance-critical data processing.
+
+This has implications when running Umbraco in Docker.
+
+For more information, refer to the [Docker documentation on storage](https://docs.docker.com/engine/storage/).
+
+### General file system consideration 
+
+In general, when working with files and Docker you work in a "push" fashion with read-only layers. When you build, you take all your files and "push" them into this read-only layer.
+
+This means that you should avoid making files on the fly, and instead rely on building your image.
+
+In an Umbraco context, this means you should not create or edit template, script or stylesheet files via the backoffice. These should be deployed as part of your web application and not managed via Umbraco. 
+
+Similarly, you shouldn't use InMemory modelsbuilder, since that also relies on creating files on the disk. While this is not a hard requirement, it doesn't provide any value if not live editing your site.
```

Comment:
```suggestion
Similarly, you shouldn't use InMemory modelsbuilder, since that also relies on creating files on the disk. While this is not a hard requirement, it doesn't provide any value unless you are live editing your site.
```